ddtrace/debugging/_probe/status.py

```diff
@@ -56,53 +73,68 @@ def _payload(
-        if self._retry_queue:
-            meter.distribution("backlog.size", len(self._retry_queue))
-
+    def _write_payload(self, payload: bytes) -> None:
         try:
-            now = time.time()
-
-            while self._retry_queue:
-                item, ts = self._retry_queue.popleft()
-                if now - ts > di_config.diagnostics_interval:
-                    # We discard the expired items as they wouldn't be picked
-                    # up by the backend anyway.
-                    continue
-
-                try:
-                    self._encoder.put(item)
-                except BufferFull:
-                    self._retry_queue.appendleft((item, ts))
-                    log.warning("Failed to clear probe status message backlog. Will try again later.")
-                    meter.increment("backlog.buffer_full")
-                    return
-
-            payload = self._payload(probe, status, message, now, error)
-
-            try:
-                self._encoder.put(payload)
-            except BufferFull:
-                log.warning("Failed to write status message to the buffer because it is full. Enqueuing for later.")
-                meter.increment("buffer_full")
-                self._retry_queue.append((payload, now))
+            log.debug("Sending probe status payload: %r", payload)
+            with self._connect() as conn:
+                conn.request(
+                    "POST",
+                    "/debugger/v1/diagnostics",
+                    payload,
+                    headers=MULTIPART_HEADERS,
+                )
+                resp = compat.get_connection_response(conn)
+                if not (200 <= resp.status < 300):
+                    log.error("Failed to upload payload: [%d] %r", resp.status, resp.read())
+                    meter.increment("upload.error", tags={"status": str(resp.status)})
+                else:
+                    meter.increment("upload.success")
+                    meter.distribution("upload.size", len(payload))
+        except Exception:
+            log.error("Failed to write payload", exc_info=True)
+            meter.increment("error")
 
+    def _enqueue(self, probe: Probe, status: str, message: str, error: t.Optional[ErrorInfo] = None) -> None:
+        self._queue.put_nowait(self._payload(probe, status, message, time.time(), error))
+        log.debug("Probe status %s for probe %s enqueued", status, probe.probe_id)
+
+    def flush(self) -> None:
+        if self._queue.empty():
+            return
+
+        msgs: t.List[str] = []
+        while not self._queue.empty():
+            msgs.append(self._queue.get_nowait())
+
+        try:
+            self._write_payload_with_backoff(
+                multipart(
+                    parts=[
+                        FormData(
+                            name="event",
+                            filename="event.json",
+                            data=f"[{','.join(msgs)}]",
```

Comment:
are there limit on number of messages we send? as this is not log backend anymore... how big the messages can be?

Reply:
On the assumption that we have a hard limit on probes, I don't think we ever risk of getting near the size limit (which I'd expect to be ~ 1 MB).